File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/cv/profile/ProfileDto.kt
package ch.streckeisen.mycv.backend.cv.profile

import ch.streckeisen.mycv.backend.cv.education.EducationDto
import ch.streckeisen.mycv.backend.cv.experience.WorkExperienceDto
import ch.streckeisen.mycv.backend.cv.profile.theme.ProfileThemeDto
import ch.streckeisen.mycv.backend.cv.project.ProjectDto
import ch.streckeisen.mycv.backend.cv.skill.SkillDto
import com.fasterxml.jackson.annotation.JsonProperty

data class ProfileDto(
    val jobTitle: String,
    val bio: String?,
    @get:JsonProperty(value = "isProfilePublic") val isProfilePublic: Boolean,
    @get:JsonProperty(value = "isEmailPublic") val isEmailPublic: Boolean,
    @get:JsonProperty(value = "isPhonePublic") val isPhonePublic: Boolean,
    @get:JsonProperty(value = "isAddressPublic") val isAddressPublic: Boolean,
    val hideDescriptions: Boolean,
    val profilePicture: String,
    val workExperiences: List<WorkExperienceDto>,
    val education: List<EducationDto>,
    val skills: List<SkillDto>,
    val projects: List<ProjectDto>,
    val theme: ProfileThemeDto?
)

File: frontend/src/dto/ProfileDto.ts
import type { WorkExperienceDto } from '@/dto/WorkExperienceDto'
import type { EducationDto } from '@/dto/EducationDto'
import type { SkillDto } from '@/dto/SkillDto'
import type { ProfileThemeDto } from '@/dto/ProfileThemeDto'
import { ProjectDto } from '@/dto/ProjectDto'

export type ProfileDto = {
  jobTitle: string
  bio: string
  isProfilePublic: boolean
  isEmailPublic: boolean
  isPhonePublic: boolean
  isAddressPublic: boolean
  hideDescriptions: boolean
  profilePicture: string
  workExperiences: Array<WorkExperienceDto>
  education: Array<EducationDto>
  skills: Array<SkillDto>
  projects: Array<ProjectDto>
  theme?: ProfileThemeDto
}
